razorpay-payments.php

```diff
@@ -292,44 +348,39 @@ function check_razorpay_response()
 
                 $success = false;
                 $error = "";
+                $captured = false;
 
                 $api = new Api($key_id, $key_secret);
-
+                
                 try
                 {
                     if ($this->payment_action === 'authorize')
                     {   
                         $payment = $api->payment->fetch($razorpay_payment_id);
-            
-                        $expectedStatus = 'authorized';
                     }
                     else
                     {
-                        $payment = $api->payment->fetch($razorpay_payment_id);
-                        $amount = $payment->amount;
+                        $razorpay_order_id = $woocommerce->session->get('razorpay_order_id');
+                        $razorpay_signature = $_POST['razorpay_signature'];
                         
-                        $payment = $payment->capture(array('amount'=>$amount)); 
-                        
-                        $expectedStatus = 'captured';
+                        $signature = hash_hmac('sha256', $razorpay_order_id . '|' . $razorpay_payment_id, $key_secret);
+
+                        if (hash_equals($signature , $razorpay_signature))
+                        {
+                            $captured = true;;
+                        }
                     }
     
                     //Check success response
-                    if ($payment->status === $expectedStatus)
+                    if ($captured)
                     {
                         $success = true;
                     }
-                    else
-                    {
+
+                    else{
                         $success = false;
```

Comment:
`$payment->amount === $amount` sounds good. I think $amount descends from the WC_Order, which makes sense.

We still need the status check though. The payment amounts might match, but we need the payment to be captured.